### DISCUSSION TITLE
Data Types from a Connected Google Sheet all Resolve to Type TEXT

### DISCUSSION THREAD

**Discussion by mcancell** [Budibase/budibase#12901] [Help]
I have linked a Sheet/table to a Data Provider in BB. Other than the Key (Rate Name) all other Columns are of type NUMBER. However, after creating the Data Provider, the Schema shows all of these as Type TEXT. AND, for some reason I cannot change the Schema (using Edit Col in BB) to Type Number. Why do my numeric Sheet Cols all get assigned Type TEXT in the Schema? And why CAN'T I change the Col Type in the BB Schema? 
![image](https://github.com/Budibase/budibase/assets/25249301/49d46bb3-312b-4a90-99ac-2b86f7610d3d)

**Comment by ConorWebb96** [comment]
Hey @mcancell,

This is a limitation of the google sheets connector. There isn't a lot we can really do on this as all data received from google sheets comes in text format. Everything will work and sync properly.

I'd recommend that you are just careful when handling your data whenever interacting with Google sheets. e.g. anything saved back will need to be converted to a string if you were to for example use a number form field.
